### ISSUE TITLE
store scrapy stats separately to full log due to potentially large file size

### ISSUE THREAD

**Issue by odscjames** [open-contracting/kingfisher-scrape#199]
From @pindec 

For some scrapes, e.g. paraguay_dncp, the log file size is very large - > 50MB.

It would be helpful for analysts to also have a separate link to the scrapy stats summary presented at the end of scraping.

Separate links should be available on:
1. the main page (log summary / full log)
2. the jobs page (log summary / full log) [note that log summary will be empty while job is running; helpful to have an interim message to say log summary not yet available in this case]
3. the logs page (list of sources and 2 logs for each source)

**Comment by odscjames** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Kingfisher-scrape - store scrapy stats separately to full log due to potentially large file size](https://trello.com/c/xEHgFO6O/114-kingfisher-scrape-store-scrapy-stats-separately-to-full-log-due-to-potentially-large-file-size)

**Comment by odscjames** [comment]
From me :smile:  : 

So the Scrapy web interface is provided by Scrapy itself; we can have a quick look but I suspect it will turn out that to modify the code will be more effort than it's worth. 

I think a first step would be to have something that extracted the summary from the full log and put it in a new file alongside the full log. It is then possible to read this from the "Logs" option in Scrapy, which allows you to browse the logs directory from a web browser. 

So basically your option 3 above, but not 1 or 2. (option 4 is reading from the console, of course)

How does that sound?

**Comment by pindec** [comment]
@odscjames thanks! option 3 is fine.

**Comment by odscjames** [comment]
I have a different suggestion to start covering this and #198 at the same time:

We add to Scrape repository a small python command line tool, "parse_scrapy_log" or something.

You call it with the path of a log file you want to parse, eg "parse_scrapy_log logs/scraper/hash.log"

It will then load the data from that log file, work out the important bits and show them as a summary.

So to start with, it would just show the "scrapy stats summary".

But for #198, we can start to build in some more explanation in the summary output.

The reason it's in the scrapy repository is we can start to tailor the output per scraper. For instance, if we are interpreting logs for one of the scrapers with authentication, we can specially flag any errors that are down to authentication problems. Or for one of the scrapers where you have to scrape listings pages then data pages, we can tell if 1 failed fetch means we are missing 1 data item or 50 and report that back.

How does that sound?

**Comment by pindec** [comment]
Thanks @odscjames . 

Sounds good. Tailored feedback for individual scrapers would be very helpful. 

Would it be possible to do this and option 3 together? So add a cron job to automatically generate the interpreted/summary stats using parse_scrapy_log which are then linked to in the scrapy logs browser?

This would mean analysts can access the interpreted stats with a click rather than logging into the server, locating the log, copying the path, looking up the command, and running it, which would be an occasional task and therefore hard to memorise.

**Comment by odscjames** [comment]
> Would it be possible to do this and option 3 together? So add a cron job to automatically generate the interpreted/summary stats using parse_scrapy_log which are then linked to in the scrapy logs browser?

Yes!

**Comment by odscjames** [comment]
Deployed and @pindec checked. I just want to add a tiny bit of docs somewhere.

**Comment by odscjames** [comment]
Docs in
